src/clr-angular/data/datagrid/datagrid-column-separator.ts

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (c) 2016-2018 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+import { DOCUMENT } from '@angular/common';
+import { Component, Inject, Renderer2 } from '@angular/core';
+
+import { ClrDragEvent } from '../../utils/drag-and-drop/drag-event';
+import { ColumnResizerService } from './providers/column-resizer.service';
+import { TableSizeService } from './providers/table-size.service';
+
+let nbCount: number = 0;
```

Comment:
Is there a reason not to use UNIQUE_ID_PROVIDER here instead?  We use it in tree-node and modals to set the instance id.

Reply:
@hippee-lee I wasn't really aware of that provider. But here we are not generating a unique id for the instance itself, but for a resize draggable's `clrGoup` input. Though I think I could still use a unique id generated by UNIQUE_ID_PROVIDER for the `clrGoup` key as it provides unique id/key. But I have one hesitation to hold off of doing so right now. So I will ask some questions below. I would love to hear you and other reviewers perspectives on the question.

We will need to generate unique id/key for draggable columns, rows., etc as well eventually. On top of that, we will have many other directives and components that need to generate some kind of unique key or id. In those cases, do we need to utilize UNIQUE_ID_PROVIDER? Because by injecting the provider, all those directives/components become dependent on one more dependency injection. Isn't it better to have fewer dependencies for directives/components if possible? And on top of that, to generate a unique id, the provider needs to get instantiated in each directive/component instances (columns, rows, cells, etc.,) while we could generate that unique id just by incrementing a integer value in the constructor. So my question is: When do we need to use UNIQUE_ID_PROVIDER? What does it do better in terms of generating a unique id/key than incrementing a integer value in the constructor?

Reply:
I think its better to have one place for generating and tracking unique ides instead of duplicating that code in different components. I think this has more value than  adding fewer dependencies to components/directives, imo. 
I believe that `useFactory` just injects the function and `let NB_INSTANCES = <value>;` is global to the Clarity app. This means one variable that stores the current `state` of uniqueness and then the function gets injected into things that need to increment it.

Reply:
I see. Thanks for the explanation. I will go with the provider then 👌